SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialRPCService.cpp

```diff
@@ -428,8 +460,37 @@ void SpatialRPCService::OnEndpointAuthorityLost(Worker_EntityId EntityId, Worker
 	}
 }
 
+void SpatialRPCService::UpdateSpanIdCache(Worker_EntityId EntityId, ERPCType Type)
+{
+	if (EventTracer == nullptr)
+	{
+		return;
+	}
+
+	EntityComponentId Id(EntityId, RPCRingBufferUtils::GetRingBufferComponentId(Type));
+	RPCRingBufferDescriptor Descriptor = RPCRingBufferUtils::GetRingBufferDescriptor(Type);
+	const RPCRingBuffer& Buffer = GetBufferFromView(EntityId, Type);
+
+	EntityRPCType EntityTypePair = EntityRPCType(EntityId, Type);
+	const uint64 LastSeenRPCId = SpanIdCache.LastSeenRPCIds.Contains(EntityTypePair) ? SpanIdCache.LastSeenRPCIds[EntityTypePair] : 0;
+	if (Buffer.LastSentRPCId >= LastSeenRPCId)
+	{
+		uint64 FirstRPCIdToRead = LastSeenRPCId + 1;
+		for (uint64 RPCId = FirstRPCIdToRead; RPCId <= Buffer.LastSentRPCId; RPCId++)
+		{
+			Schema_FieldId FieldId = Descriptor.GetRingBufferElementFieldId(RPCId);
+			Trace_SpanId SpanId = EventTracer->GetSpanId(Id, FieldId);
```

Comment:
Do we wanna log in case `EventTracer` doesn't give us a valid span ID here (or is null span id a valid case)?